src/server/api/memobase_server/controllers/event.py

```
@@ -119,3 +150,59 @@ async def update_user_event(
         user_event.event_data = new_events
         session.commit()
     return Promise.resolve(None)
+
+
+async def search_user_events(
+    user_id: str,
+    project_id: str,
+    query: str,
+    topk: int = 10,
+    similarity_threshold: float = 0.5,
+    time_range_in_days: int = 7,
+) -> Promise[UserEventsData]:
```

Since CONFIG.enable_event_embedding is now available, I think we should add a check here, right?

---

If not enable, all the embeddings are none and `pg-vector` will just return a empty list.

Should we raise an error for this API while not enable event embedding?

---

yeah, raising an error makes sense.